cache/versioned_test.go

```diff
@@ -31,5 +37,14 @@ func TestVersioned(t *testing.T) {
 		assert.Equal(t, v1Data, resV1)
 		resV2 := v2.Fetch(context.Background(), []string{"hit", "miss"})
 		assert.Equal(t, v2Data, resV2)
+
+		err := v1.Delete(context.Background(), "hit")
+		require.NoError(t, err)
+		resV1 = v1.Fetch(context.Background(), []string{"hit", "miss"})
+		assert.Equal(t, map[string][]uint8{}, resV1)
+		err = v2.Delete(context.Background(), "hit")
```

Comment:
I think here you want to test that V1 doesn't delete V2 too. So I think it's more interesting calling `v2.Fetch()` and ensuring it still returns a value. Then you can try to delete V2 too.